pallets/dapp-staking/src/types.rs

```diff
@@ -993,35 +1001,98 @@ impl EraInfo {
     }
 }
 
+/// Bonus status to track remaining 'voting stake' safe move actions in the ongoing B&E subperiod.
+/// Move actions during B&E refer either to:
+/// - a 'partial unstake with voting stake decrease',
+/// - a 'stake transfer between two contracts'.
+#[derive(Encode, Decode, MaxEncodedLen, Copy, Clone, Debug, TypeInfo)]
+#[scale_info(skip_type_params(MaxBonusMoves))]
+pub enum BonusStatus<MaxBonusMoves: Get<u8>> {
+    /// Bonus rewards are forfeited.
+    BonusForfeited,
+    /// Bonus rewards are preserved with a value which is the number of remaining 'safe moves' allowed in the ongoing B&E subperiod.
+    SafeMovesRemaining(u8),
+    #[codec(skip)]
+    _Phantom(PhantomData<MaxBonusMoves>),
+}
```

Comment:
At the moment we have issues with MBM migrations, that's why we didn't go with async backing on Astar yet.

If this approach is adopted, we'll be blocked with the feature until an adequate fix is provided for that issue.

However, taking the approach with additional enum value, `OneMoveRemaining` solves the need for any migration whatsoever. Only the "storage version" needs to be bumped.

EDIT:
Or even better - why not just pure `u8` here (ofc, it can be wrapped into a struct)? Call the variable "moves_remaining" and that's it?